event/bus_test.go

```diff
@@ -12,82 +12,93 @@ package event
 
 import (
 	"sync"
-	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestBus_IfPublishingWithoutSubscribers_ItShouldNotBlock(t *testing.T) {
-	bus := NewBus(0, 0)
+type BusTestSuite struct {
```

Comment:
praise: Whilst if we get serious about testing this package long term I might not want us to test it like this, this is a very nice and low risk way of testing multiple implementations without the cost of setting up an action based int. test system for the package, and it keeps any would-be-effort largely out of the rod-code PR.

Thanks Keenan :)